File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Response/IssueResponse.java
package ut.edu.be_quanlytro.Dto.Response;

import lombok.Builder;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
public class IssueResponse {
    private UUID id;
    private String roomNumber;
    private String tenantName;
    private String description;
    private String imageUrl;
    private String status;
    private String solutionNote;
    private LocalDateTime createdAt;
}

File: Flutter_QuanLyTro/lib/data/models/response/issue_response.dart
class IssueResponse {
  final String id;
  final String roomNumber;
  final String tenantName;
  final String description;
  final String? imageUrl;
  final String status;
  final String? solutionNote;
  final DateTime? createdAt;

  IssueResponse({
    required this.id,
    required this.roomNumber,
    required this.tenantName,
    required this.description,
    this.imageUrl,
    required this.status,
    this.solutionNote,
    this.createdAt,
  });

  factory IssueResponse.fromJson(Map<String, dynamic> json) {
    return IssueResponse(
      id: json['id'] ?? '',
      roomNumber: json['roomNumber'] ?? '',
      tenantName: json['tenantName'] ?? '',
      description: json['description'] ?? '',
      imageUrl: json['imageUrl'],
      status: json['status'] ?? 'PENDING',
      solutionNote: json['solutionNote'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
    );
  }
}